snowflake/src/main/java/org/apache/iceberg/snowflake/JdbcSnowflakeClient.java

```diff
@@ -343,4 +366,32 @@ public SnowflakeTableMetadata loadTableMetadata(SnowflakeIdentifier tableIdentif
   public void close() {
     connectionPool.close();
   }
+
+  private Optional<RuntimeException> snowflakeExceptionToIcebergException(
```

Comment:
I think you could avoid the optional with all the additional `orElseGet` statements above by just passing in the generic text and falling back to an UncheckedSQLException.